tests/unit/mainwindow/test_prompt.py

```diff
@@ -94,6 +96,33 @@ def test_backspacing_path(self, qtbot, tmp_path, get_prompt):
         prompt.item_focus('next')
         assert prompt._lineedit.text() == str(testdir / 'bar')
 
+    @pytest.mark.parametrize("keys,expected",
+                            [([], ['bar', 'foo', 'bat'].sort()),
+                            (['F'], ['foo'].sort()),
+                            (['Backspace', 'A'], ['bar', 'bat'].sort())])
+    def test_filtering_path(self, qtbot, tmp_path, get_prompt, keys, expected):
+        testdir = tmp_path / 'test'
+
+        for directory in ['bar', 'foo', 'bat']:
+            (testdir / directory).mkdir(parents=True)
+
+        prompt = get_prompt(str(testdir) + os.sep)
+
+        for _ in range(len(keys)):
```

Comment:
I don't understand this. Why do you want to run the entire test multiple times based on how many keys there are?